cmd/account/mgmt/account-iam.go

```diff
@@ -0,0 +1,152 @@
+package mgmt
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/openshift/osdctl/internal/utils/globalflags"
+	osdCloud "github.com/openshift/osdctl/pkg/osdCloud"
+	awsprovider "github.com/openshift/osdctl/pkg/provider/aws"
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+)
+
+// iamOptions defines the struct for running the iam command
+type iamOptions struct {
+	awsAccountID string
+	awsProfile   string
+	awsRegion    string
+	kerberosUser string
+	rotate       bool
+}
+
+var arnPolicy = "arn:aws:iam::aws:policy/AdministratorAccess"
+
+// accountIamCmd implements the accountIam command which creates an IAM user for a given account
+func newCmdAccountIAM(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags, globalOpts *globalflags.GlobalOptions) *cobra.Command {
+	ops := &iamOptions{}
+	iamCmd := &cobra.Command{
+		Use:               "iam",
+		Short:             "Creates an IAM user in a given AWS account and prints out the credentials",
+		Args:              cobra.NoArgs,
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(ops.complete(cmd, args))
+			cmdutil.CheckErr(ops.run())
+		},
+	}
+
+	iamCmd.Flags().StringVarP(&ops.awsAccountID, "accountId", "i", "", "AWS account ID to run this against")
+	iamCmd.Flags().StringVarP(&ops.kerberosUser, "user", "u", "", "Kerberos username to run this for")
+	iamCmd.Flags().StringVarP(&ops.awsProfile, "profile", "p", "", "AWS Profile")
+	iamCmd.Flags().StringVarP(&ops.awsRegion, "region", "r", "", "AWS Region")
+	iamCmd.Flags().BoolVarP(&ops.rotate, "rotate", "R", false, "Rotate an IAM user's credentials and print the output")
+
+	return iamCmd
+}
+
+func (o *iamOptions) complete(cmd *cobra.Command, args []string) error {
+	if o.awsAccountID == "" {
+		return cmdutil.UsageErrorf(cmd, "AWS Account ID must be provided")
+	}
+	if o.kerberosUser == "" {
+		return cmdutil.UsageErrorf(cmd, "Kerberos UID is required")
+	}
+	if o.awsRegion == "" {
+		o.awsRegion = "us-east-1"
+	}
+	if o.awsProfile == "" {
+		o.awsProfile = "osd-staging-2"
+	}
+	return nil
+}
+
+func checkError(msg string, err error) {
+	if err != nil {
+		r := fmt.Errorf(msg, err)
+		panic(r)
+	}
+}
+
+func writeFile(user string, content string) {
+	data := []byte(content)
+	fileName := fmt.Sprintf("/tmp/aws-iam-%s", user)
+	err := os.WriteFile(fileName, data, 0644)
+	checkError("Failed to create credential file in /tmp", err)
+	fmt.Printf("AWS credentials file created in %s\n", fileName)
+}
+
+func (o *iamOptions) run() error {
+	// Build the base AWS client using the provide credentials (profile or env vars)
+	awsClient, err := awsprovider.NewAwsClient(o.awsProfile, o.awsRegion, "")
+	checkError("Could not build AWS Client: %s", err)
```

Comment:
I thought about using a function to handle the errors to try to avoid multiple blocks like the one below and keep the code cleaner:
``` 
if err != nil {
  return fmt.Errorf("could not build AWS Client: %s", err)
}
````
I guess using an if statement for handling the errors is the preferable approach @mjlshen ?

Reply:
Yeah you're right, I forgot about that :D

Reply:
I have changed this in my latest commit https://github.com/openshift/osdctl/pull/383/commits/7e4074638b2f762d11de73a80d318cc915f03821